ch_pipeline/processing/daily.py

```diff
@@ -713,41 +716,38 @@ def _available_tags(self):
 
         This includes any that currently exist or are in the job queue.
         """
-
-        csds = []
-
-        # For each interval find and add all CSDs that have not already been added
-        for interval in self._intervals:
-            csd_i = csds_in_range(*interval)
-            csd_set = set(csds)
-            csds += [csd for csd in csd_i if csd not in csd_set]
-
-        # grab the list of files that are online, and that exist anywhere, from the earliest csd to the latest
+        # Lets us get only unique csds in the order we want them
+        # with a single pass.
+        csds = unique_ordered(
+            # This is a generator
+            (csd for i in self._intervals for csd in csds_in_range(*i))
+        )
         csds_sorted = sorted(csds)
+
+        # grab the list of files that are online, and that exist anywhere,
+        # from the earliest csd to the latest
         filenames_online, filenames_that_exist = self._available_files(
             csds_sorted[0], csds_sorted[-1] + 1
         )
-
         # only queue jobs for which all data is available online in chime_online
         csds_available = self._csds_available_data(
```

Comment:
I haven't worked much with sets in Python, so apologies if this is a basic question, but: in changing `_csds_available_data()` to return a set instead of a list, is there a danger that the ordering of `tags` could be unintentionally different from the ordering of `csds_sorted`, because the latter gets converted to a set within `_csds_available_data()`? Or does this not happen, or not matter?

Reply:
In theory, a set should still maintain the order that items were added to it. However, this shouldn't actually matter here since we're pulling the csd order from `csds` rather than `csds_available`, so the order of `csds_available` won't matter. The reason I changed it to a set is because the lookup for whether an item exists in a set is O(1) on average, whereas it's O(n) for a list, so there's a marginal performance boost. The order in `csds` will be based on the revconfig file, so that's what we want to keep.

Reply:
Ah of course, I overlooked the line where `csds_available` is simply used as a filter to select from the (ordered) elements of `csds`. Thanks!